posts/posts-router.js

```diff
@@ -0,0 +1,95 @@
+const express = require('express');
+
+const Helpers = require('../data/db.js');
+
+const router = express.Router();
+
+//POST request
+router.post('/', async (req, res) => {
+    try {
+        
+        if (req.body.title == null || req.body.contents == null) {
+            res.status(400).json({errorMessage: "Please provide title and contents for the post."});
+        } else {
+            const post = await Helpers.insert(req.body);
+            res.status(201).json(post)
+        }
+    } catch (error) {
+        console.log(error);
+        res.status(500).json({ error: "There was an error while saving the post to the database"});
+    }
+})
+
+//GET request
+router.get('/', async (req, res) => {
+    try {
+        const posts = await Helpers.find(req.query);
+        res.status(200).json(posts);
+    } catch (error) {
+        console.log(error);
+        res.status(500).json({error: "The posts information could not be retrieved." });
+    }
+});
+
+//GET Post by ID
+router.get('/:id', async (req, res) => {
+    try {
+        const post = await Helpers.findById(req.params.id);
+        if (post.id == req.params.id) {
+            res.status(200).json(post);
+        } else {
+            res.status(404).json({message: "The post with the specified ID does not exist."});
+        }
+    } catch (error) {
+        console.log(error);
+        res.status(500).json({error: "The post information could not be retrieved."});
+    }
+});
+
+//DELETES a post, but returns that deleted object
+router.delete('/:id', async (req, res) => {
+    try {
+        const post = await Helpers.findById(req.params.id);
+        if (post.id != req.params.id) {
+            res.status(404).json({message: "The post with the specified ID does not exist."});
+        }
+    }  catch {
+          res.status(500).json({error: "The post could not be removed"});
+    }
+    try{
+        const post = await Helpers.findById(req.params.id);
+                     
+        const destroy = await Helpers.remove(req.params.id);
+        if (destroy) {
+            res.status(200).json(post);
+        }
+    } catch (error) {
+        console.log(error);
+        res.status(500).json({error: "The post could not be removed"});
+    }
```

Comment:
I encountered an error here, and I think it is related to the logic of your delete function. While I received an error message in Postman saying that the post with the specified ID doesn't exist, the delete successfully went through, which leads me to believe that your logic for your delete's error checking is faulty somehow.

Another error involves line 64, `UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)`

A possible solution for this could be to nest these try functions instead of running them one after another. Another suggestion for the other error would be to carefully check your variable declarations.

Reply:
This is fixed